alacritty_terminal/src/event_loop.rs

```diff
@@ -259,15 +259,18 @@ where
                 for event in events.iter() {
                     match event.key {
                         tty::PTY_CHILD_EVENT_TOKEN => {
-                            if let Some(tty::ChildEvent::Exited) = self.pty.next_child_event() {
+                            if let Some(tty::ChildEvent::Exited(code)) = self.pty.next_child_event()
+                            {
                                 if self.hold {
                                     // With hold enabled, make sure the PTY is drained.
                                     let _ = self.pty_read(&mut state, &mut buf, pipe.as_mut());
                                 } else {
                                     // Without hold, shutdown the terminal.
                                     self.terminal.lock().exit();
                                 }
-
+                                if let Some(code) = code {
+                                    self.event_proxy.send_event(Event::ExitCode(code));
+                                }
```

Comment:
I think this should be not `ExitCode`, but `ChildExit` or something like that and then we follow it with e.g `exit`, so your sending should be above, since right now you send

```
Exit,
ExitCode
```

Which is a bit _weird_, also you may want to know the exit code for `--hold`, but then `ExitCode` is weird, because it could mean that terminal exited with that code.

Reply:
I've used `ChildExit`.